### PR TITLE
PR: Remove Stripe Checkout (the legacy modal)

### LINKED ISSUE #785: Remove Legacy Checkout (Modal)
Right now, in the Stripe settings, the merchant can choose between having the payment fields in the checkout page (the default), or using "Stripe Checkout", which opens a <del>completely separate page (similar to PayPal)</del> modal.

We should check if the Stripe Checkout works correctly with Payment Intents, and if it's not easy to make it work, consider removing it.

Edit: When removing the Legacy Checkout altogether, we'll need to remember to remove the warning notice added here: https://github.com/woocommerce/woocommerce-gateway-stripe/pull/837/files#diff-bb6de50012b6937b8a786bcb0f70bdd7R119

### PR DESCRIPTION
**Pull request by RadoslavGeorgiev**
Fixes #785

#### Changes proposed in this Pull Request:

Based on our discussions with Stripe and SCA-related work, and as described in @allendav's [comment](https://github.com/woocommerce/woocommerce-gateway-stripe/issues/785#issuecomment-481456778), we will remove support for Stripe Checkout in 4.2.0

This PR removes all traces of Stripe Checkout, including:

1. The main part of the code, located in `stripe.js` and `WC_Gateway_Stripe`, which handles the modal
2. All necessary settings, their docs (.md files), and the admin, which controls the visibility of those fields
3. Translations
4. Screenshot from the WP.org readme
5. Admin tests
6. Support for charges (instead of intents) in `WC_Gateway_Stripe`. This should not affect any extensions, since `WC_Gateway_Stripe::process_payment` is only responsible for one-time and initial payments.

#### Testing instructions

1. Navigate to the gateway's settings and ensure that there are no Stripe Checkout-related settings.
2. Do a few smoke tests with test payments, and make sure nothing blows up.

### CODE REVIEW

File: includes/class-wc-gateway-stripe.php [hunk 1]
```diff
@@ -485,194 +442,45 @@ public function payment_scripts() {
...
-		$stripe_params['inline_cc_form']                          = $this->inline_cc_form ? 'yes' : 'no';
-		$stripe_params['stripe_checkout_require_billing_address'] = apply_filters( 'wc_stripe_checkout_require_billing_address', false ) ? 'yes' : 'no';
-		$stripe_params['is_checkout']                             = ( is_checkout() && empty( $_GET['pay_for_order'] ) ) ? 'yes' : 'no'; // wpcs: csrf ok.
-		$stripe_params['return_url']                              = $this->get_stripe_return_url();
-		$stripe_params['ajaxurl']                                 = WC_AJAX::get_endpoint( '%%endpoint%%' );
-		$stripe_params['stripe_nonce']                            = wp_create_nonce( '_wc_stripe_nonce' );
-		$stripe_params['statement_descriptor']                    = $this->statement_descriptor;
-		$stripe_params['elements_options']                        = apply_filters( 'wc_stripe_elements_options', array() );
-		$stripe_params['sepa_elements_options']                   = apply_filters(
+		$sepa_elements_options = apply_filters(
```
Comment (jrodger) [inline includes/class-wc-gateway-stripe.php]: Does moving the creation of these options up to here relate to removing Stripe Checkout? Or is it just a bit of tidy up? Possibly to do with lining up equals signs?

Not a problem either way, just want to make sure I'm not missing anything related to Stripe Checkout removal.

File: includes/abstracts/abstract-wc-stripe-payment-gateway.php [hunk 2]
```diff
@@ -690,7 +690,6 @@ public function prepare_source( $user_id, $force_save_source = false ) {
 		}
 
 		return (object) array(
-			'is_intent'     => false,
```
Comment (jrodger) [inline includes/abstracts/abstract-wc-stripe-payment-gateway.php]: This was added as part of the SCA work back in a1d897a1527f14f2a799283301ccb72aaacc775b. I haven't followed through to exactly what it's being used for, but do we want to remove it here?

Comment (RadoslavGeorgiev) [reply includes/abstracts/abstract-wc-stripe-payment-gateway.php]: This removal is not legacy-modal related, but while I was doing cleanup I found out that this flag was still in place.

It was supposed to be used when we were supporting non-PI and PI-based credit card payments within the gateway and the reason for that actually used to be Stripe Checkout. Now everything is PI-based, so the flag/prop is not needed anymore.

File: includes/class-wc-gateway-stripe.php (hunk 1 above)
Comment (RadoslavGeorgiev) [reply includes/class-wc-gateway-stripe.php]: Yes, I'm working with a sniffer enabled and wanted to align the equal signs/tidy up. Nothing more :)

**Comment by jrodger** [comment]
Thanks for clarifying those few points. All looks good.

:shipit:

**Pull request merged**